### PR TITLE
optional xhr property added to dropzone DropzoneFile

### PR DESCRIPTION
**Pull request by Hikariii**
Please fill in this template.

- [x] Prefer to make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/enyo/dropzone/blob/master/src/dropzone.coffee#L1136-L1137
- [x] Increase the version number in the header if appropriate.
Is already the latest version...

### CODE REVIEW

**Comment by outring** [comment]
Hi, @Hikariii! Thank you for the fix. Could you also write a test for this, please?
You could just add a check for some XHR property in `file.xhr` near the line 150 in dropzone-tests.ts and do the same with dropzone-commonjs-tests.ts near the line 144

**Comment by Hikariii** [comment]
Hey @outring, when reading the instructions http://definitelytyped.org/guides/contributing.html#tests I see that tests do not contain assertions, just code which is type-checked.

>The tests are not runnable in the TDD way but contain code that should compile with no errors, usually taken from the documentation samples of the library.

I made double checked that the upload code is called from the tests and that is the case already.
The function `processFiles` and  `uploadFile` both call the `uploadFiles` method that adds the xhr object.
So not really shure what to add in a test if the code is already tested.

**Comment by outring** [comment]
@Hikariii, you're right, code is not runnable, but you still need to use the `.xhr` and something inside of it, to make sure that `.xhr` property is in the right place (i.e. in `DropzoneFile`) and that it's actually the `XMLHttpRequest` instance.
As you can see in https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/dropzone/dropzone-tests.ts#L150 we're checking that `DropzoneFile` has `.accepted` property, and also has `.previewElement` which is HTMLElement (by accessing  `.classList`).
So basically you can check that `f.xhr.readyState` is there, for example.

**Comment by outring** [comment]
Looks great! Thanks!

👍  for merge
